python/lancedb/table.py

```diff
@@ -521,6 +521,52 @@ def update(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def cleanup_old_versions(
+        self,
+        older_than: Optional[timedelta] = None,
+        *,
+        delete_unverified: bool = False,
+    ) -> CleanupStats:
+        """
+        Clean up old versions of the table, freeing disk space.
+
+        Note: This function is not available in LanceDb Cloud (since LanceDb
+        Cloud manages cleanup for you automatically)
+
+        Parameters
+        ----------
+        older_than: timedelta, default None
+            The minimum age of the version to delete. If None, then this defaults
+            to two weeks.
+        delete_unverified: bool, default False
+            Because they may be part of an in-progress transaction, files newer
+            than 7 days old are not deleted by default. If you are sure that
+            there are no in-progress transactions, then you can set this to True
+            to delete all files older than `older_than`.
+
+        Returns
+        -------
+        CleanupStats
+            The stats of the cleanup operation, including how many bytes were
+            freed.
+        """
+
+    @abstractmethod
+    def compact_files(self, *args, **kwargs):
+        """
+        Run the compaction process on the table.
+
+        Note: This function is not available in LanceDb Cloud (since LanceDb
+        Cloud manages compaction for you automatically)
+
+        This can be run after making several small appends to optimize the table
+        for faster reads.
+
+        Arguments are passed onto :meth:`lance.dataset.DatasetOptimizer.compact_files`.
```

Comment:
Should check me on this, but I think since we use mkdocs not Sphinx, method references look like:
```suggestion
        Arguments are passed onto [lance.dataset.DatasetOptimizer.compact_files][].
```

Reply:
Yep.  Good catch.

Reply:
I've removed the link entirely (since it is now an external link from LanceDB docs -> Lance docs and wasn't sure if we wanted a potentially unstable `https://...` link)

Reply:
I'm not sure if Github is just being slow / showing me stale files or if I'm missing something obvious: https://github.com/lancedb/lancedb/pull/900/commits/16aafe9427f5d53332462a05ef4f36ed22dd3ba6